core/src/broadcast_stage/standard_broadcast_run.rs

```diff
@@ -344,7 +344,7 @@ impl StandardBroadcastRun {
         if now - last > BROADCAST_PEER_UPDATE_INTERVAL_MS
             && self
                 .last_peer_update
-                .compare_and_swap(now, last, Ordering::Relaxed)
+                .compare_and_swap(last, now, Ordering::Relaxed)
                 == last
         {
             *self.cluster_nodes.write().unwrap() = ClusterNodes::<BroadcastStage>::new(
```

Comment:
according to this metrics, this iteration takes 2-3 ms (excluding spikes) on testnet. so there should be immediate need for async (separate thread/service):

![image](https://user-images.githubusercontent.com/117807/126686394-eecd2afc-ccb4-4c37-a15b-d03fd0b4ed31.png)

code: https://github.com/solana-labs/solana/pull/18776/files#r675045904